include/seqan3/core/simd/view_iota_simd.hpp

```diff
@@ -0,0 +1,248 @@
+    counted_simd_iterator(counted_simd_iterator const &) noexcept = default; //!< Defaulted.
+    counted_simd_iterator(counted_simd_iterator &&) noexcept = default; //!< Defaulted.
+    counted_simd_iterator & operator=(counted_simd_iterator const &) noexcept = default; //!< Defaulted.
+    counted_simd_iterator & operator=(counted_simd_iterator &&) noexcept = default; //!< Defaulted.
+    ~counted_simd_iterator() = default; //!< Defaulted.
+
+    /*!\brief Constructs and initialises the iterator with the given index.
+     *
+     * \tparam scalar_index_t The scalar index type; must model seqan3::arithmetic.
+     *
+     * \param[in] scalar_index The scalar index the iterator shall represent.
+     */
+    template <arithmetic scalar_index_t>
+    explicit counted_simd_iterator(scalar_index_t const scalar_index) noexcept :
+        count_simd{simd::fill<index_t>(scalar_index)},
+        count_scalar{static_cast<size_t>(scalar_index)}
+    {}
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+
+    //!\brief Access the pointed-to matrix coordinate column.
+    reference operator*() const
+    {
+        return count_simd;
+    }
+    //!\}
+
+    /*!\name Arithmetic operators
+     * \{
+     */
+
+    //!\brief Increments the iterator.
+    counted_simd_iterator & operator++()
+    {
+        count_simd += seqan3::simd::fill<index_t>(1);
+        ++count_scalar;
+        return *this;
+    }
+
+    //!\brief Increments the iterator and returns the iterator pointing to the previous index.
+    counted_simd_iterator operator++(int)
+    {
+        counted_simd_iterator tmp{*this};
+        ++(*this);
+        return tmp;
+    }
+
+    //!\brief Returns the distance between two iterators.
+    difference_type operator-(counted_simd_iterator const & rhs) const
+    {
+        return count_scalar - rhs.count_scalar;
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+
+    //!\brief Tests whether `lhs == rhs`.
+    friend bool operator==(counted_simd_iterator const & lhs, counted_simd_iterator const & rhs) noexcept
+    {
+        return lhs.count_scalar == rhs.count_scalar;
+    }
+
+    //!\brief Tests whether `lhs != rhs`.
+    friend bool operator!=(counted_simd_iterator const & lhs, counted_simd_iterator const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\brief A closed iota view over a simd vector.
+ * \ingroup simd
+ * \implements std::ranges::forward_range
+ * \implements std::ranges::sized_range
+ * \implements std::ranges::common_range
+ * \implements std::ranges::borrowed_range
+ *
+ * \tparam index_t The represented index type; must model seqan3::simd::simd_concept.
+ *
+ * \details
+ *
+ * This view is a lightweight wrapper around a seqan3::detail::counted_simd_iterator pair.
+ * The regular std::views::iota view cannot be used in combination with a simd vector type because the comparison of
```

Comment:
```suggestion
 *
 * Note:
 * The regular std::views::iota view cannot be used in combination with a simd vector type because the comparison of
```

Reply:
BTW what do you mean by this?